typescript/infra/test/govern.hardhat-test.ts

```diff
@@ -133,14 +133,15 @@ describe('ICA governance', async () => {
     contracts = await new InterchainAccountDeployer(multiProvider).deploy(
       routerConfig,
     );
-    // local = contracts[localChain].interchainAccountRouter;
+    local = contracts[localChain].interchainAccountRouter;
     remote = contracts[remoteChain].interchainAccountRouter;
     icaApp = new InterchainAccount(contracts, multiProvider);
 
     accountConfig = {
       origin: TestChainName.test1,
       owner: signer.address,
-      localRouter: remote.address,
+      localRouter: local.address,
+      routerOverride: remote.address,
```

Comment:
is this honored by `deployAccount`? https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/10f6495878528d2a73cd3b4824c1a74bca3b2fba/typescript/sdk/src/middleware/account/InterchainAccount.ts#L89-L102

The interface is a bit awkward, idk why there are overrides in the `router` but also there are positional params to that function. Seems like the config overrides are pretty much ignored?